pandas/computation/align.py

```diff
@@ -172,12 +172,8 @@ def _reconstruct_object(typ, obj, axes, dtype):
         ret_value = res_t.type(obj)
     else:
         ret_value = typ(obj).astype(res_t)
+        if len(obj.shape) == 1 and len(obj) == 1:
```

Comment:
Thanks, but we can't use ``isscalar`` to distinguish scalar and 1 element array (e.g. ``np.array(0)`` and ``np.array([0])``).

```
np.array(0).shape
# ()
lib.isscalar(np.array(0).item())
# True

np.array([0]).shape
# (1,)
lib.isscalar(np.array([0]).item())
# True
```